### PR TITLE
Adds Info and Debug logger functionality

### LINKED ISSUE #252: Add Support for BP_LOG_LEVEL in `scribe`
There should be some mechanism added to `scribe` that would respect various level of log verbosity in accordance with [RFC 0027](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0027-log-levels.md). 

As it stands `scribe` has no way of tagging certain logs to certain levels of verbosity. As outlined in the RFC `scribe` needs to respect two levels of verbosity set using the `BP_LOG_LEVEL` environment variable. Those two levels are which are:
- `INFO` (which is the default logging level) which logs additional information about the build process (i.e. notes on the decision making process of build)
- `DEBUG` this includes all logs that are produced for the `INFO` level as well as further information that would aid in diagnosing and debug build process problems that do not result in a failure (i.e. full logs of build tools)

### PR DESCRIPTION
**Pull request by ForestEckhardt**
Resolves #252 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by fg-j** [comment]
This is awesome. Could you add some godoc examples of how a buildpack author can incorporate this logger into their buildpack in practice? The tests sort of show it, but a direct example would be great.

**Comment by ForestEckhardt** [comment]
For sure I could doc up the whole package if that is something we are looking for!

**Comment by ryanmoran** [comment]
There's quite a bit of duplication happening here. We've now got 3 different loggers with duplicative implementation APIs and switching logic. I think we can achieve something a bit simpler by just controlling whether or not the writer goes somewhere or is discarded. I've thrown together a simplified example here: https://github.com/paketo-buildpacks/packit/blob/log-levels-alt/scribe/logger.go

Take a look and let me know what you think.

**Comment by ForestEckhardt** [comment]
Moving this to draft until I have the examples written up!

**Comment by ForestEckhardt** [comment]
@fg-j @ryanmoran Alrighty took me a little longer than I thought but the docs are up and running. Couple call outs.
- Ryan and I have decided to deprecate the Bar object as it is not fully implemented and we do not foresee it being used in the near future and and we are going to use it we would like to overhaul the interface.
- I could not get a tested example working for the Color object because of that I also do not put together examples for the writer. I could make an indented writer but it did not seem completely necessary to me to set up examples. If you would like an example just let me know.
- I just put together the best blurb that I could for the top level description I am more than happy to get feedback on this one.

Hope this accomplishes everything that y'all are looking for.

File: scribe/logger.go [hunk 1]
```diff
@@ -6,7 +6,45 @@ import (
 	"strings"
 )
 
+// A Logger provides a standard logging interface for doing basic low level
+// logging task as well as debug logging.
```
Comment (fg-j) [inline scribe/logger.go:10]:
```suggestion
// logging tasks as well as debug logging.
```

File: scribe/logger.go [hunk 2]
```diff
@@ -6,7 +6,45 @@ import (
...
 type Logger struct {
+	writer io.Writer
+	LeveledLogger
+	Debug LeveledLogger
+}
+
+// NewLogger take a writer and return a Logger that writes to the given
```
Comment (fg-j) [inline scribe/logger.go:17]:
```suggestion
// NewLogger takes a writer and returns a Logger that writes to the given
```

File: scribe/scribe.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Package scribe provies a set of interfaces to allow buildpack authors to
+// control their logs on varying levels of granularity. This exposes high level
+// logging functions with packit specific business logic baked into the logging
+// as well as low level functionality that allows an author to build up their
+// own functionality using atomic peices.
```
Comment (fg-j) [inline scribe/scribe.go:5]:
```suggestion
// own functionality using atomic pieces.
```

File: scribe/writer.go [hunk 4]
```diff
@@ -5,28 +5,38 @@ import (
 	"io"
 )
 
+// An Option is a way to configure a writers format.
 type Option func(Writer) Writer
 
+// WithColor take a Color and returns an Option which can be passed in while
```
Comment (fg-j) [inline scribe/writer.go:11]:
```suggestion
// WithColor takes a Color and returns an Option which can be passed in while
```

**Review by fg-j — CHANGES_REQUESTED** [review]
Maybe I didn't understand your comment @ForestEckhardt about getting a working example with Color, but if possible I'd love to see an example of how to properly use the logger with Debug functionality.

File: scribe/logger.go [hunk 5]
```diff
@@ -6,7 +6,45 @@ import (
...
+func NewLogger(writer io.Writer) Logger {
+	return Logger{
+		writer:        writer,
+		LeveledLogger: NewLeveledLogger(writer),
+		Debug:         NewLeveledLogger(io.Discard),
+	}
+}
+
+// WithLevel takes in a log level string configures the log level of the
+// logger. To enable debug logging the log level must be set to "DEFAULT".
```
[the remark this answers is not in the document]
Comment (fg-j) [reply scribe/logger.go:28]: ^^ If `DEFAULT` is in fact the correct word, then this docstring confused me!

[the remark this answers is not in the document]
Comment (ForestEckhardt) [reply scribe/logger.go:28]: Whoops

**Comment by ForestEckhardt** [comment]
> Maybe I didn't understand your comment @ForestEckhardt about getting a working example with Color, but if possible I'd love to see an example of how to properly use the logger with Debug functionality.

My trouble with Color is that the only way I know haw to test the examples is by testing the contents of stdout. The Colors function adds a prefix and a suffix that is interpreted by the terminal and changes the display but seemingly does not make it to standard out. If there is another way of getting the examples working and testing it I am game!

I totally forgot that was the feature I added to start this work 🤦  I will add an example ASAP!

**Comment by ForestEckhardt** [comment]
@fg-j So after so discussion with @ryanmoran and a little investigating on his part he discovered that: 
> The terminal escape sequences are dropped if the output is not a TTY, and the io.Pipe that the Examples feature uses is not a TTY. 

So it would appear that it is not possible to test the Color functionality using the Go Examples testing framework. I hope that clears things up.

**Review by fg-j — APPROVED** [review]
Thanks for adding that debug logging example. LGTM!

File: scribe/logger.go [hunk 6]
```diff
@@ -6,7 +6,45 @@ import (
...
+		writer:        writer,
+		LeveledLogger: NewLeveledLogger(writer),
+		Debug:         NewLeveledLogger(io.Discard),
+	}
+}
+
+// WithLevel takes in a log level string and configures the log level of the
+// logger. To enable debug logging the log level must be set to "DEBUG".
+func (l Logger) WithLevel(level string) Logger {
+	switch level {
```
Comment (ryanmoran) [inline scribe/logger.go:30]:
```suggestion
	switch strings.ToUpper(level) {
```

**Review by ryanmoran — APPROVED** [review]

**Review by fg-j — APPROVED** [review]

**Pull request merged**